### PR TITLE
Correlation id is missing from header

### PR DESCRIPTION
**Pull request by lasrinnil**
Description
-------------
This PR will introduce `CorrelationIdMiddleware` to help identify the CorrelationId when needed. To begin with, helping the TimeSeriesBundleIngestionEndpoint HttpRespons contain the CorrelationId for the request. And thereby solving the referred issue down below.

https://app.zenhub.com/workspaces/joules-5fd22a1a59e4740018f5ab5a/issues/energinet-datahub/green-energy-hub/270

### CODE REVIEW

File: source/TimeSeries/IntegrationTests/TimeSeriesBundleIngestionEndpointTests.cs [hunk 1]
```diff
@@ -58,6 +58,7 @@ public async Task When_RequestReceivedWithJwtToken_Then_AcceptedResponseReturned
             using var request = await CreateTimeSeriesHttpRequest(true, content).ConfigureAwait(false);
             var response = await Fixture.HostManager.HttpClient.SendAsync(request).ConfigureAwait(false);
             response.StatusCode.Should().Be(HttpStatusCode.Accepted);
+            response.Headers.Contains("CorrelationId").Should().BeTrue();
```
Comment (BjarkeMeier) [inline source/TimeSeries/IntegrationTests/TimeSeriesBundleIngestionEndpointTests.cs:61]: This assertion doesn't fit the test method name.
I think this should be a separate test.

File: source/TimeSeries/UnitTests/Infrastructure/Correlation/CorrelationContextTests.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using Energinet.DataHub.Core.TestCommon.AutoFixture.Attributes;
+using Energinet.DataHub.TimeSeries.Infrastructure.Correlation;
+using Xunit;
+
+namespace Energinet.DataHub.TimeSeries.UnitTests.Infrastructure.Correlation
+{
+    public class CorrelationContextTests
```
Comment (BjarkeMeier) [inline source/TimeSeries/UnitTests/Infrastructure/Correlation/CorrelationContextTests.cs:23]: Do we really want to maintain all these tests, middleware, and more as duplicates in more domains?

File: source/TimeSeries/Infrastructure/Correlation/TraceContext.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        public string TraceId { get; }
+
+        public string ParentId { get; }
+
+        public bool IsValid { get; }
+
+        public static TraceContext Parse([NotNull] string traceContext)
+        {
+            if (string.IsNullOrWhiteSpace(traceContext)) return Invalid();
+            if (traceContext.Length != 55) return Invalid();
```
Comment (johevemi) [inline source/TimeSeries/Infrastructure/Correlation/TraceContext.cs:45]: Magic numbers - should they be variables or something? I'm not sure what the numbers mean or where they are from

File: source/TimeSeries/IntegrationTests/TimeSeriesBundleIngestionEndpointTests.cs (hunk 1 above)
Comment (lasrinnil) [reply source/TimeSeries/IntegrationTests/TimeSeriesBundleIngestionEndpointTests.cs:61]: We have a lot of integration tests with a very narrow scope. But sure, whats one more :)

Comment (BjarkeMeier) [reply source/TimeSeries/IntegrationTests/TimeSeriesBundleIngestionEndpointTests.cs:61]: I find that a very good thing.
And hiding assertions in other tests makes it very hard to get an overview of what is actually being tested and what is not.

File: source/TimeSeries/Infrastructure/Correlation/TraceContext.cs (hunk 3 above)
Comment (lasrinnil) [reply source/TimeSeries/Infrastructure/Correlation/TraceContext.cs:45]: I will add a comment why in the code :)

But it is the totalt required length of the tracecontext, for more info please see here:

https://www.w3.org/TR/trace-context/#version-format

File: source/TimeSeries/UnitTests/Infrastructure/Correlation/CorrelationContextTests.cs (hunk 2 above)
Comment (lasrinnil) [reply source/TimeSeries/UnitTests/Infrastructure/Correlation/CorrelationContextTests.cs:23]: Of course not! I have been talking to Volt and they would use this in a package if I created one, which I will. But to begin with I will implement this locally in TimeSeries domain, which hopefully will help our actors when sending in messages. Then I will create a package of `CorrelationIdMiddelware` which actually is used locally in MeteringPoint, Charges, and TimeSeries.

**Review by BjarkeMeier — APPROVED** [review]

**Review by johevemi — APPROVED** [review]

**Pull request merged**